blobstore/blobstore.go

```diff
@@ -199,12 +201,40 @@ func ParseUpload(req *http.Request) (blobs map[string][]*BlobInfo, other url.Val
 			return nil, nil, err
 		}
 		bi.BlobKey = appengine.BlobKey(params["blob-key"])
+		charset := params["charset"]
+
 		if ctype != "message/external-body" || bi.BlobKey == "" {
 			if formKey != "" {
 				slurp, serr := ioutil.ReadAll(part)
 				if serr != nil {
 					return nil, nil, errorf("error reading %q MIME part", formKey)
 				}
+
+				// Handle base64 content transfer encoding. multipart.Part transparently
+				// handles quoted-printable, and no special handling is required for
+				// 7bit, 8bit, or binary.
+				ctype, params, err = mime.ParseMediaType(part.Header.Get("Content-Transfer-Encoding"))
+				if err == nil && ctype == "base64" {
```

Comment:
lowercase ctype for this check.

> These values are not case sensitive. That is, Base64 and BASE64 and bAsE64 are all equivalent.

via https://www.w3.org/Protocols/rfc1341/5_Content-Transfer-Encoding.html

Reply:
mime.ParseMediaType handles lowercasing it. https://golang.org/pkg/mime/#ParseMediaType

> On success, ParseMediaType returns the media type converted to lowercase and trimmed of white space and a non-nil map.